features/collection/api/src/main/java/org/opennms/netmgt/collection/support/IndexStorageStrategy.java

```diff
@@ -65,7 +66,7 @@ public final String getResourceTypeName() {
     @Override
     public String getResourceNameFromIndex(CollectionResource resource) {
         // Use the instance value as the name of the resource
-        return resource.getInstance();
+        return GenericTypeResource.sanitizeInstance(resource.getInstance());
```

Comment:
One concern here is that the various strategies now need to explicitly sanitize the instance, making it easy to forget or omit when creating new strategies.

Are there any other places where we need to add calls to `sanitizeInstance`?

Reply:
Yeah, it's not easy to do without verifying all of the strategies individually. I agree that it's not ideal but moving that to the persistence layer is a larger effort.

I think a less intrusive approach to solving this would be to make the unsanitized instance available as a new getter and update to the `ObjectNameStorageStrategy` to use that. This would leave the existing code path untouched and maintain the current behavior.